natural-sql-7b/model/model.py

```diff
@@ -0,0 +1,58 @@
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+
+class Model:
+    def __init__(self):
+        self.tokenizer = None
+        self.model = None
+
+    def load(self):
+        """
+        Initializes the AutoTokenizer and AutoModelForCausalLM with the 'chatdb/natural-sql-7b' model.
+        """
+        self.tokenizer = AutoTokenizer.from_pretrained("chatdb/natural-sql-7b")
+        self.model = AutoModelForCausalLM.from_pretrained(
+            "chatdb/natural-sql-7b",
+            device_map="auto",
+            torch_dtype=torch.float16,
+        )
+
+    def predict(self, questions):
+        """
+        Generates SQL queries from a list of questions.
+
+        :param questions: List of natural language questions.
+        :return: List of SQL queries corresponding to the input questions.
+        """
```

Comment:
I like that its documenting each function. It can help our users understand the code a bit quicker.